src/main/java/com/oracle/nosql/spring/data/core/NosqlTemplateBase.java

```diff
@@ -116,6 +121,8 @@ protected boolean doCreateTableIfNotExists(
         NosqlEntityInformation<?, ?> entityInformation) {
 
         String tableName = entityInformation.getTableName();
+        String autogen = getAutoGenType(entityInformation);
```

Comment:
This method should do 2 things: check if table exists and call either of the 2 other methods: doCreateTable() or doCheckExistingTable().

Reply:
Changed. Split into getCreateTableDDL(), doCreateTable() and doCheckExistingTable()